packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -49,6 +52,8 @@ typedef RouteBuilder<T> = Route<T> Function(BuildContext context, RouteSettings
 /// Used by [Navigator.onGenerateInitialRoutes].
 typedef RouteListFactory = List<Route<dynamic>> Function(NavigatorState navigator, String initialRoute);
 
+typedef RestorableRouteBuilder<T> = Route<T> Function(BuildContext context, Object arguments);
```

Comment:
docs

Reply:
Interestingly, the analyzer didn't warn about the missing docs here. Filed https://github.com/dart-lang/linter/issues/2238.